ebike_city_tools/optimize/round_optimized_sort_selection.py

```diff
@@ -0,0 +1,224 @@
+import math
+import pandas as pd
+import networkx as nx
+import numpy as np
+
+from ebike_city_tools.optimize.linear_program import define_IP
+from ebike_city_tools.optimize.round_simple import edge_to_source_target
+from ebike_city_tools.utils import (
+    lane_to_street_graph,
+    extend_od_circular,
+    compute_car_time,
+    compute_edgedependent_bike_time,
+    output_to_dataframe,
+)
+from ebike_city_tools.iterative_algorithms import transform_car_to_bike_edge
+from ebike_city_tools.metrics import compute_travel_times_in_graph
+
+FLOW_CONSTANT = 1
+
+def sort_by_bikevalue(capacities) :
+    """Sorts the capacities by planned bike capacity, with ties broken by car capacity."""
+    return capacities.sort_values(["u_b(e)", "u_c(e)"], ascending=[False, True])
+
+def rounding_error(fractional_value):
+    """Given a fractional value, returns the distance to the closest integer."""
+    next_int = round(fractional_value)
+    if fractional_value >= 0:
+        return 0
+    return abs(next_int - fractional_value)
+
+def rounding_error_of_row(row):
+    """Auxiliary funtion determining the rounding error of the car capacity."""
+    return rounding_error(row["u_c(e)"]) #+ rounding_error(row["u_c(e)_reversed"])
+
+def round_row(row):
+    """Given a row, this function returns the row obtained by rounding the car and bike capacities."""
+    # print(row)
+    new_value = min(1, round(row["u_c(e)"]))
+    # new_value = round(row["u_c(e)"])
+    rounded_by = row["u_c(e)"] - new_value
+    row["u_c(e)"] = new_value
+    row["u_b(e)"] = row["u_b(e)"] + rounded_by
+    return row
+
+def sort_by_rounding_error(capacities) :
+    """Sorts the capacities by increasing rounding error of the car capacity."""
+    cap_with_rounding_error = capacities
+    cap_with_rounding_error["rounding_error"] = capacities.apply(rounding_error_of_row, axis=1)
```

Comment:
Could probably be accelarated by doing something like 
```
capacities["rounded_car"] = capacities["u_c(e)"].round()
cap_with_rounding_error["rounding_error"] = (capacities["u_c(e)"] - capacities["rounded_car"]).abs()
```
(.apply is usually slower than column-wise operations)

Reply:
Sounds great! I do not know, how to write good pandas code ^^'

Reply:
I can make such changes after we merge the code